tfhe/src/shortint/engine/server_side/mul.rs

```diff
@@ -58,6 +63,11 @@ impl ShortintEngine {
         ct_left: &mut Ciphertext,
         ct_right: &Ciphertext,
     ) -> EngineResult<()> {
+        if ct_left.degree == Degree(0) || ct_right.degree == Degree(0) {
```

Comment:
ct_left.degree.0 == 0 ?

Reply:
A matter of taste I guess 😄 we used to access the content of the new type usually, having to type NewType(raw_value) is a bit cumbersome don't you think ?

or maybe we need some additional partial eq/eq partial ord/ord that can work with a usize as rhs ?

Reply:
Having impls of comparisons for Degree with a usize would diminish the purpose of having the Degree newtype

Deriving ord/partialor so that we can compare a Degree to another Degree makes sense and would allow to type the .0 in places where we have degree on both side (e lhs.degree <= rhs.degree)

 as for ct_left.degree.0 == 0 vs ct_left.degree.0 == Degree(0), yeah its a matter of taste

Reply:
dominant style which I prefer is the one on the left to not have to important the potentially missing new type 🙂

Reply:
ok